model_compression_toolkit/core/common/quantization/node_quantization_config.py

```diff
@@ -140,15 +143,17 @@ def fln_quantization(self):
         return self.quant_mode == ActivationQuantizationMode.FLN_QUANT
 
     def set_activation_quantization_param(self,
-                                          activation_params: dict):
+                                          activation_params: dict,
+                                          node: BaseNode):
         """
          Set a quantization parameter for the node's activation.
 
         Args:
             activation_params: Dictionary that contains weight quantization params.
+            node: node in a graph that represents the model.
 
         """
-        assert self.quant_mode == ActivationQuantizationMode.QUANT
+        assert node.is_activation_quantization_enabled() or node.is_fln_quantization()
```

Comment:
no need to add node to the function's arguments, the node's check `is_fln_quantization` is actually inspecting the node's candidates to decide if this an FLN. This function `set_activation_quantization_param` is already a candidate class function, so you can just check the class `quant_mode`